server/events/command_runner.go

```diff
@@ -134,55 +134,89 @@ func (c *DefaultCommandRunner) RunAutoplanCommand(baseRepo models.Repo, headRepo
 		return
 	}
 
-	projectCmds, err := c.ProjectCommandBuilder.BuildAutoplanCommands(ctx)
+	// Run plan command for all projects
+	planErr := c.runAutoCommand(ctx, models.PlanCommand, AutoplanCommand{})
+
+	if planErr == nil {
+		// Run policy_check command
+		_ = c.runAutoCommand(ctx, models.PolicyCheckCommand, AutoPolicyCheckCommand{})
+	}
+}
+
+func (c *DefaultCommandRunner) runAutoCommand(ctx *CommandContext, cmdModel models.CommandName, pullCommand PullCommand) error {
+	var projectCmds []models.ProjectCommandContext
+	var err error
+
+	switch cmdModel.String() {
+	case "plan":
+		projectCmds, err = c.ProjectCommandBuilder.BuildAutoplanCommands(ctx)
+	case "policy_check":
+		projectCmds, err = c.ProjectCommandBuilder.BuildAutoPolicyCheckCommands(ctx)
+	}
+
 	if err != nil {
-		if statusErr := c.CommitStatusUpdater.UpdateCombined(ctx.Pull.BaseRepo, ctx.Pull, models.FailedCommitStatus, models.PlanCommand); statusErr != nil {
+		if statusErr := c.CommitStatusUpdater.UpdateCombined(ctx.Pull.BaseRepo, ctx.Pull, models.FailedCommitStatus, cmdModel); statusErr != nil {
 			ctx.Log.Warn("unable to update commit status: %s", statusErr)
 		}
 
-		c.updatePull(ctx, AutoplanCommand{}, CommandResult{Error: err})
-		return
+		c.updatePull(ctx, pullCommand, CommandResult{Error: err})
+		return err
 	}
+
 	if len(projectCmds) == 0 {
-		log.Info("determined there was no project to run plan in")
+		ctx.Log.Info("determined there was no project to run %s in", cmdModel.String())
 		if !c.SilenceVCSStatusNoPlans {
 			// If there were no projects modified, we set a successful commit status
 			// with 0/0 projects planned successfully because some users require
 			// the Atlantis status to be passing for all pull requests.
 			ctx.Log.Debug("setting VCS status to success with no projects found")
-			if err := c.CommitStatusUpdater.UpdateCombinedCount(baseRepo, pull, models.SuccessCommitStatus, models.PlanCommand, 0, 0); err != nil {
+			if err := c.CommitStatusUpdater.UpdateCombinedCount(ctx.Pull.BaseRepo, ctx.Pull, models.SuccessCommitStatus, cmdModel, 0, 0); err != nil {
 				ctx.Log.Warn("unable to update commit status: %s", err)
 			}
 		}
-		return
+		return err
 	}
```

Comment:
imo this is kind of odd right, because presumably this should be resolved for plan, since policy check always takes place after plan during the auto command.

Maybe this is a more high level PR comment, but after looking through your PR, there is a lot of duplication which is the way it is between plan and apply commands as well. However for policy check, i think of it similar as init, except that it will always follow plan.  So the question I want to know is would it be cleaner if we kept it as a step in the plan command instead? ie. here: https://github.com/lyft/atlantis/blob/ORCA-470-add-policy-check-step-runner/server/events/yaml/valid/global_cfg.go#L83:L85

Benefits I see are the following:
* you don't need to add a separate `parallel*` flag since it will be handled as part of the plan
* you won't be running through unnecessary logic multiple times (building the project config and all that. For projects that are skipping the clone step, it's two downloads of the repo config file.
* A lot less code being written

The only disadvantage I see is that this will make it harder to create a comment command for policy checks specifically, though imo this is ok since it should always follow plan.